test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/lang/ClassJSTest.scala

```diff
@@ -19,6 +19,47 @@ import scala.scalajs.js
 
 class ClassJSTest {
 
+  @Test def isAssignableFrom(): Unit = {
+    /* isAssignableFrom should respect the JVM rules even for JS types,
+     * although isInstance doesn't.
```

Comment:
I think this warrants a short sentence of *why*. (FWIW, I agree with this, because `isAssignableFrom` basically allows reflection on the class hierarchy, so the *declared* classes matter).